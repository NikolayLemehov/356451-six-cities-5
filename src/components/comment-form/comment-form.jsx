import React, {PureComponent} from "react";
import PropTypes from "prop-types";

const RADIO_VALUES = [`5`, `4`, `3`, `2`, `1`];

class CommentForm extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {rating, review, onSubmit, onFieldChange} = this.props;

    return (
      <form className="reviews__form form" action="#" method="post"
        onSubmit={onSubmit}
      >
        <label className="reviews__label form__label" htmlFor="review">Your review</label>
        <div className="reviews__rating-form form__rating">
          {RADIO_VALUES.map((it) => (
            <React.Fragment key={it}>
              <input className="form__rating-input visually-hidden" name="rating" value={it} id={`${it}-stars`}
                onChange={onFieldChange} type="radio" checked={rating === it}
              />
              <label htmlFor={`${it}-stars`} className="reviews__rating-label form__rating-label" title="perfect">
                <svg className="form__star-image" width="37" height="33">
                  <use xlinkHref="#icon-star"/>
                </svg>
              </label>
            </React.Fragment>
          ))}
        </div>
        <textarea className="reviews__textarea form__textarea" id="review" name="review"
          placeholder="Tell how was your stay, what you like and what can be improved"
          onChange={onFieldChange} value={review} />
        <div className="reviews__button-wrapper">
          <p className="reviews__help">
            To submit review please make sure to set <span className="reviews__star">rating</span> and
            describe your stay with at least <b className="reviews__text-amount">50 characters</b>.
          </p>
          <button className="reviews__submit form__submit button" type="submit" disabled="">Submit</button>
        </div>
      </form>
    );
  }
}

CommentForm.propTypes = {
  rating: PropTypes.string.isRequired,
  review: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onFieldChange: PropTypes.func.isRequired,
};

export default CommentForm;
