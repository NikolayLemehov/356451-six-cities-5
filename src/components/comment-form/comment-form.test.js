import React from "react";
import renderer from "react-test-renderer";
import {CommentForm} from "./comment-form";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({adapter: new Adapter()});

const noop = () => {};
describe(`CommentForm should render correctly`, () => {
  it(`CommentForm should render correctly by default`, () => {
    const tree = renderer
      .create(<CommentForm
        offerId={`1`}
        uploadReviewAction={noop}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`CommentForm should render correctly when checked twoStarsInput`, () => {
    const tree = mount(
        <CommentForm
          offerId={`1`}
          uploadReviewAction={noop}
        />
    );

    const twoStarsInput = tree.find(`.form__rating-input`).at(3);
    twoStarsInput.simulate(`change`);
    expect(tree.debug()).toMatchSnapshot();
  });
});
