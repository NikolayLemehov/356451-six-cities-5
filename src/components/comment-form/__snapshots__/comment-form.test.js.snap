// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm should render correctly CommentForm should render correctly by default 1`] = `
<form
  action="#"
  className="reviews__form form"
  method="post"
  onSubmit={[Function]}
>
  <label
    className="reviews__label form__label"
    htmlFor="review"
  >
    Your review
  </label>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="5-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="4-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="3-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="2-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="1-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    id="review"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value=""
  />
  <div
    className="reviews__button-wrapper"
  >
    <p
      className="reviews__help"
    >
      To submit review please make sure to set 
      <span
        className="reviews__star"
      >
        rating
      </span>
       and describe your stay with at least 
      <b
        className="reviews__text-amount"
      >
        50 characters
      </b>
      .
    </p>
    <button
      className="reviews__submit form__submit button"
      disabled="disabled"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`CommentForm should render correctly CommentForm should render correctly when checked twoStarsInput 1`] = `
"<CommentForm offerId=\\"1\\" uploadReviewAction={[Function: noop]}>
  <form className=\\"reviews__form form\\" action=\\"#\\" method=\\"post\\" onSubmit={[Function: handleSubmit]}>
    <label className=\\"reviews__label form__label\\" htmlFor=\\"review\\">
      Your review
    </label>
    <div className=\\"reviews__rating-form form__rating\\">
      <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"5\\" id=\\"5-stars\\" onChange={[Function (anonymous)]} type=\\"radio\\" checked={false} />
      <label htmlFor=\\"5-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\">
        <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
          <use xlinkHref=\\"#icon-star\\" />
        </svg>
      </label>
      <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"4\\" id=\\"4-stars\\" onChange={[Function (anonymous)]} type=\\"radio\\" checked={false} />
      <label htmlFor=\\"4-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\">
        <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
          <use xlinkHref=\\"#icon-star\\" />
        </svg>
      </label>
      <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"3\\" id=\\"3-stars\\" onChange={[Function (anonymous)]} type=\\"radio\\" checked={false} />
      <label htmlFor=\\"3-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\">
        <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
          <use xlinkHref=\\"#icon-star\\" />
        </svg>
      </label>
      <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"2\\" id=\\"2-stars\\" onChange={[Function (anonymous)]} type=\\"radio\\" checked={true} />
      <label htmlFor=\\"2-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\">
        <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
          <use xlinkHref=\\"#icon-star\\" />
        </svg>
      </label>
      <input className=\\"form__rating-input visually-hidden\\" name=\\"rating\\" value=\\"1\\" id=\\"1-stars\\" onChange={[Function (anonymous)]} type=\\"radio\\" checked={false} />
      <label htmlFor=\\"1-stars\\" className=\\"reviews__rating-label form__rating-label\\" title=\\"perfect\\">
        <svg className=\\"form__star-image\\" width=\\"37\\" height=\\"33\\">
          <use xlinkHref=\\"#icon-star\\" />
        </svg>
      </label>
    </div>
    <textarea className=\\"reviews__textarea form__textarea\\" id=\\"review\\" name=\\"review\\" placeholder=\\"Tell how was your stay, what you like and what can be improved\\" onChange={[Function (anonymous)]} value=\\"\\" />
    <div className=\\"reviews__button-wrapper\\">
      <p className=\\"reviews__help\\">
        To submit review please make sure to set 
        <span className=\\"reviews__star\\">
          rating
        </span>
         and describe your stay with at least 
        <b className=\\"reviews__text-amount\\">
          50 characters
        </b>
        .
      </p>
      <CommentFormSubmitButton isDisabledSubmitButton={false}>
        <button className=\\"reviews__submit form__submit button\\" type=\\"submit\\" disabled=\\"disabled\\">
          Submit
        </button>
      </CommentFormSubmitButton>
    </div>
  </form>
</CommentForm>"
`;
